### PR TITLE
Add version endpoint

### PR DESCRIPTION
**Pull request by Retler**
### Description
This PR adds a version endpoint to the encryption server. At the moment the version will include the git commit hash and the git tag (if the commit is tagged). Other service versions can be added later to the protobuf VersionResponse message.

### Parent Issue
Issue #73 

### Comments for the Reviewers
Do we want to include more versioning information at the moment?

### Type(s) of Change (Split the PR if you check many)
- [ ] Added user feature
- [x] Added technical feature
- [x] Added tests
- [ ] Refactor
- [ ] Bugfix
- [ ] Updated documentation
- [ ] Infrastructure changes (Terraform, GCP, K8s, other)
- [ ] CI/CD workflow changes

### Testing Checklist
- [x] I have added/updated unit tests for functional changes.
- [ ] I have added/updated integration tests for changes that affect dependent systems.
- [x] I have added/updated end-to-end tests for feature changes.

### General Checklist
- [x] I have pulled in the latest changes from master.
- [x] I have run `make lint`.
- [x] I have successfully run `make docker-up; make tests`.
- [x] I have updated relevant workflows and deployment tools.
- [ ] I have updated/added relevant documentation (readme, doc comments, etc). (N/A, setting version happens automatically)
- [x] I have added the license to new source code files.
- [x] I have spent some time looking over the full diff before creating this PR.

### CODE REVIEW

File: encryption-service/tests/grpc_e2e_tests/client.go [hunk 1]
```diff
@@ -149,3 +149,14 @@ func (c *Client) CreateUser(usertype app.CreateUserRequest_UserKind) (*app.Creat
 	}
 	return createUserResponse, nil
 }
+
+// Perform a `CreateUser` request.
```
Comment (bruno-cc) [inline encryption-service/tests/grpc_e2e_tests/client.go:153]: This comment seems wrong

Comment (Retler) [reply encryption-service/tests/grpc_e2e_tests/client.go]: Ah yes.. Fixed

**Review by bruno-cc — DISMISSED** [review]
Looks good! Thank you.

**Comment by philip-cc** [comment]
Remember to update the API spec and maybe also the user manual.

**Review by bruno-cc — APPROVED** [review]

**Pull request closed without merging**